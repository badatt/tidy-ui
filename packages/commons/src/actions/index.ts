import { PageActionTypes } from './page.actions';
import { RootActionTypes } from './root.actions';

/**
 * Tidy UI components internal actions
 */
export type AppActionTypes = PageActionTypes | RootActionTypes;

export { PageActions } from './page.actions';
export { RootActions } from './root.actions';
