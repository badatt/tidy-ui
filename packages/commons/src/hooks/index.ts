export { default as useColor } from './useColor';
export { default as useEscaper } from './useEscaper';
export { default as useHash } from './useHash';
export { default as useIsMounted } from './useIsMounted';
export { default as useKeyDown } from './useKeyDown';
export { default as useKeyPress } from './useKeyPress';
export { default as usePortal } from './usePortal';
export { default as useTheme } from './useTheme';
export { default as useTimeout } from './useTimeout';
export { default as useWindow } from './useWindow';
