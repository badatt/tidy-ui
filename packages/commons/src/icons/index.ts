export { default as AddCircleIcon } from './AddCircleIcon';
export { default as AddIcon } from './AddIcon';
export { default as CancelIcon } from './CancelIcon';
export { default as CheckCircleIcon } from './CheckCircleIcon';
export { default as CloseIcon } from './CloseIcon';
export { default as DangerousIcon } from './DangerousIcon';
export { default as ExpandLessIcon } from './ExpandLessIcon';
export { default as ExpandMoreIcon } from './ExpandMoreIcon';
export * from './Icon';
export { default as InfoIcon } from './InfoIcon';
export { default as LaunchIcon } from './LaunchIcon';
export { default as LinkIcon } from './LinkIcon';
export { default as RemoveIcon } from './RemoveIcon';
export { default as RotatingCircleIcon } from './RotatingCircleIcon';
export { default as WarningIcon } from './WarningIcon';
