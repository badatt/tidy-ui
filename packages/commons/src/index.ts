export * from './actions';
export * from './hooks';
export * from './icons';
export * from './provider';
export * from './reducers';
export * from './theme';
export * from './utils';
