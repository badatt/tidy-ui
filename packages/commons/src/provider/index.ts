export * from './context';
export * from './TidyUiProvider';
