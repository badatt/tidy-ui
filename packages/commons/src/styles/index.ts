export { default as font } from './font';
