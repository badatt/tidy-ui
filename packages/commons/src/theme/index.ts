import baseStyled, { css as baseCss, ThemedCssFunction, ThemedStyledInterface } from 'styled-components';

export { color } from './colors';
export * from './utils';

import { ITidyUITheme } from './types';
export * from './types';
const styled = baseStyled as ThemedStyledInterface<ITidyUITheme>;
const css = baseCss as ThemedCssFunction<ITidyUITheme>;

export { css, styled };

export { defaultBase as baseTheme } from './base';
export { theme as orchidDark } from './orchid-dark';
export { theme as orchidLight } from './orchid-light';
