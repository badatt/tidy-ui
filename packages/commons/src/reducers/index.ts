import { AppActionTypes } from '../actions';
import pageReducer from '../reducers/page.reducer';
import rootReducer from '../reducers/root.reducer';
import { ITidyUITheme, orchidDark, orchidLight } from '../theme';
import { storage } from '../utils';
import { IState } from './types';

/**
 * State initializer
 *
 * @param {IState} initState init state
 * @returns {IState} initial state
 */
const initializer = (initState: IState): IState => {
  return { ...initState };
};

const initialState: IState = {
  pageState: {},
  rootState: {},
};

/**
 * Create initial state
 *
 * @param {ITidyUITheme} theme creates initial state
 * @returns {IState} initial state
 */
const createInitialState = (theme?: ITidyUITheme): IState => {
  const themeName = storage.get('tidyui-theme');
  let baseTheme: ITidyUITheme;
  if (themeName) {
    switch (themeName) {
      case orchidLight.name:
        baseTheme = orchidLight;
        break;
      case orchidDark.name:
        baseTheme = orchidDark;
        break;
      default:
        baseTheme = theme ?? orchidLight; // Default to orchidLight if no theme matches
    }
  } else {
    baseTheme = theme ?? orchidLight; // Default to orchidLight if no theme is provided
  }
  return {
    ...initialState,
    pageState: {
      theme: baseTheme,
    },
  };
};

/**
 * Main reducer
 *
 * @param {IState} state application state
 * @param {AppActionTypes} actions all application actions
 * @returns {IState} mutated state
 */
const mainReducer = (state: IState, actions: AppActionTypes): IState => {
  const { pageState, rootState } = state;
  return {
    pageState: pageReducer(pageState, actions),
    rootState: rootReducer(rootState, actions),
  };
};

export { createInitialState, initializer, initialState, mainReducer };
