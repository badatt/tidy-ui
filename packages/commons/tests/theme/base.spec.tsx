/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { baseTheme } from '../../src';

describe('Base theme', () => {
  it('Basic', () => {
    expect(baseTheme).toMatchSnapshot();
  });
});
