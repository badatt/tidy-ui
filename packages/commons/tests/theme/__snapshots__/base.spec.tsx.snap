// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base theme Basic 1`] = `
Object {
  "font": Object {
    "bold": 700,
    "family": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", Oxygen, Cantarell, sans-serif\`",
    "light": 200,
    "medium": 400,
    "mono": "ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace",
    "regular": 500,
    "size": "1rem",
  },
  "layout": Object {
    "radius": "0.5rem",
    "screens": Object {
      "lg": Object {
        "breakpoint": 1200,
      },
      "md": Object {
        "breakpoint": 992,
      },
      "sm": Object {
        "breakpoint": 768,
      },
      "xl": Object {
        "breakpoint": 1400,
      },
      "xs": Object {
        "breakpoint": 576,
      },
    },
    "shadow": "rgb(255, 255, 255) 0px 0px 0px 0px, rgba(15, 23, 42, 0.05) 0px 0px 0px 1px, rgba(0, 0, 0, 0.1) 0px 20px 25px -5px, rgba(0, 0, 0, 0.1) 0px 8px 10px -6px",
  },
  "typography": Object {
    "body1": Object {
      "fontSize": "1rem",
      "fontWeight": 400,
      "letterSpacing": "0.00938em",
      "lineHeight": 1.5,
      "marginBottom": "0.5rem",
    },
    "body2": Object {
      "fontSize": "0.875rem",
      "fontWeight": 400,
      "letterSpacing": "0.01071em",
      "lineHeight": 1.43,
      "marginBottom": "0.4375rem",
    },
    "caption": Object {
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "letterSpacing": "0.03333em",
      "lineHeight": 1.66,
      "marginBottom": "0.375rem",
    },
    "h1": Object {
      "fontSize": "6rem",
      "fontWeight": 300,
      "letterSpacing": "-0.01562em",
      "lineHeight": 1.167,
      "marginBottom": "3rem",
    },
    "h2": Object {
      "fontSize": "3.75rem",
      "fontWeight": 300,
      "letterSpacing": "-0.00833em",
      "lineHeight": 1.2,
      "marginBottom": "1.875rem",
    },
    "h3": Object {
      "fontSize": "3rem",
      "fontWeight": 400,
      "letterSpacing": "0em",
      "lineHeight": 1.167,
      "marginBottom": "1.5rem",
    },
    "h4": Object {
      "fontSize": "2.125rem",
      "fontWeight": 400,
      "letterSpacing": "0.00735em",
      "lineHeight": 1.235,
      "marginBottom": "1.0625rem",
    },
    "h5": Object {
      "fontSize": "1.5rem",
      "fontWeight": 400,
      "letterSpacing": "0em",
      "lineHeight": 1.334,
      "marginBottom": "0.75rem",
    },
    "h6": Object {
      "fontSize": "1.25rem",
      "fontWeight": 500,
      "letterSpacing": "0.0075em",
      "lineHeight": 1.6,
      "marginBottom": "0.325rem",
    },
    "hero": Object {
      "fontSize": "6rem",
      "fontWeight": 300,
      "letterSpacing": "-0.01562em",
      "lineHeight": 1.167,
      "marginBottom": "3rem",
    },
    "p": Object {
      "fontSize": "1rem",
      "fontWeight": 400,
      "letterSpacing": "0.00938em",
      "lineHeight": 1.5,
      "marginBottom": "0.5rem",
    },
    "span": Object {
      "fontSize": "0.875rem",
      "fontWeight": 400,
      "letterSpacing": "0.01071em",
      "lineHeight": 1.43,
      "marginBottom": "0.4375rem",
    },
    "subtitle1": Object {
      "fontSize": "2.125rem",
      "fontWeight": 400,
      "letterSpacing": "0.00735em",
      "lineHeight": 1.235,
      "marginBottom": "1.0625rem",
    },
    "subtitle2": Object {
      "fontSize": "1.5rem",
      "fontWeight": 400,
      "letterSpacing": "0em",
      "lineHeight": 1.334,
      "marginBottom": "0.75rem",
    },
    "title1": Object {
      "fontSize": "3.75rem",
      "fontWeight": 300,
      "letterSpacing": "-0.00833em",
      "lineHeight": 1.2,
      "marginBottom": "1.875rem",
    },
    "title2": Object {
      "fontSize": "3rem",
      "fontWeight": 400,
      "letterSpacing": "0em",
      "lineHeight": 1.167,
      "marginBottom": "1.5rem",
    },
  },
}
`;
